core/src/main/scala/spinal/core/Component.scala

```diff
@@ -137,6 +137,12 @@ abstract class Component extends NameableByComponent with ContextUser with Scala
     children.foreach(_.walkComponents(body))
   }
 
+  val areas = ArrayBuffer[Area]()
+  def walkAreas(body : Area => Unit) : Unit = {
+    areas.foreach(_.walkAreas(body))
+    children.foreach(_.walkAreas(body))
```

Comment:
I think children means component here, this is not area. 
The name walkAreas seems misleading here.

Reply:
> The idea here is to match walkComponents, so we don't want to just walk all the areas in the component, but also all the areas recursively in all the child components too.

I can understand that, how about leave them standalone?

Reply:
I think it'd be confusing if `walkAreas` had the same naming convention and calling style of `walkComponents` but only returned the current components areas. I can just remove this function altogether for now if that helps make the PR more streamlined.